OpenRA.Mods.Common/Widgets/Logic/Ingame/GameInfoStatsLogic.cs

```diff
@@ -97,17 +97,20 @@ public GameInfoStatsLogic(Widget widget, World world, OrderManager orderManager,
 
 					var flag = item.Get<ImageWidget>("FACTIONFLAG");
 					flag.GetImageCollection = () => "flags";
+
+					var factionName = pp.DisplayFaction.Name;
 					if (player == null || player.RelationshipWith(pp) == PlayerRelationship.Ally || player.WinState != WinState.Undefined)
 					{
 						flag.GetImageName = () => pp.Faction.InternalName;
-						var factionName = pp.Faction.Name != pp.DisplayFaction.Name ? $"{pp.DisplayFaction.Name} ({pp.Faction.Name})" : pp.Faction.Name;
-						item.Get<LabelWidget>("FACTION").GetText = () => factionName;
+						factionName = pp.Faction.Name != factionName ? $"{factionName} ({pp.Faction.Name})" : pp.Faction.Name;
 					}
 					else
-					{
 						flag.GetImageName = () => pp.DisplayFaction.InternalName;
-						item.Get<LabelWidget>("FACTION").GetText = () => pp.DisplayFaction.Name;
-					}
+
+					var factionLabel = item.Get<LabelWithTooltipWidget>("FACTION");
+					var truncFactionName = WidgetUtils.TruncateText(factionName, factionLabel.Bounds.Width, Game.Renderer.Fonts[factionLabel.Font]);
+					factionLabel.GetText = () => truncFactionName;
+					WidgetUtils.TruncateLabelToTooltip(factionLabel, factionName);
```

Comment:
The truncation and assignment of `GetText` is happening in `WidgetUtils.TruncateLabelToTooltip` so you can reduce this to just:
```csharp
WidgetUtils.TruncateLabelToTooltip(item.Get<LabelWithTooltipWidget>("FACTION"), factionName);
```